internal/objectstore/dynamic_cache.go

```diff
@@ -453,33 +419,16 @@ func (d *DynamicCache) gvrFromKey(ctx context.Context, key store.Key) (schema.Gr
 	_, span := trace.StartSpan(ctx, "dynamicCache:gvrFromKey")
 	defer span.End()
 
-	// short circuit for CRD type
+	// short circuit for CRD type - RESTmapper will convert this to /,Resource= otherwise
 	if key.APIVersion == "apiextensions.k8s.io/v1" && key.Kind == "CustomResourceDefinition" {
 		return schema.GroupVersionResource{Group: "apiextensions.k8s.io", Version: "v1", Resource: "customresourcedefinitions"}, nil
 	}
 
 	gk := key.GroupVersionKind().GroupKind()
 	var gvr schema.GroupVersionResource
-
-	v, ok := d.gvrCache.Load(gk)
-	if !ok {
-		gvr, _, err := d.client.Resource(gk)
-		if err != nil {
-			return schema.GroupVersionResource{}, err
-		}
-
-		if gvr.Resource == "" {
-			_, gvr = meta.UnsafeGuessKindToResource(key.GroupVersionKind())
-		}
-
-		// do not store an empty GVR
-		if gvr.Version == "" && gvr.Resource == "" {
-			return schema.GroupVersionResource{}, fmt.Errorf("unable to locate GVR for GVK: %s", key.GroupVersionKind())
-		}
-
-		d.gvrCache.Store(gk, gvr)
-	} else {
-		gvr, _ = v.(schema.GroupVersionResource)
+	gvr, _, err := d.client.Resource(gk)
```

Comment:
How do we feel about creating a static list, like we have for GVKs that makes the Key to a GVR for known types and attempts to look it up for unknown types?

Reply:
Turns out the special case for CRDs was due to improperly passing the version to RESTMapper.

We probably don't need this except for testing via https://github.com/kubernetes/client-go/blob/master/dynamic/fake/simple.go#L43